File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    counter = {}
    
    for t in tangerine:
        if t in counter:
            counter[t] +=1
        else:
            counter[t] = 1
    freq = sorted(counter.values(),reverse=True)
    
    for f in freq:
        if k <= 0:
            break
        k-=f
        answer+=1
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let answer = 0;
    let total = 0;
    let arr = {}
    for(let i=0; i<tangerine.length; i++){
            arr[tangerine[i]] = (arr[tangerine[i]]||0)+1;
        }
    let counts = Object.values(arr).sort((a,b)=>b-a);
    for(let count of counts){
        answer+=1;
        total += count;
        if(k <= total) break;            
    }
    return answer;
}